arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -856,6 +873,16 @@ class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdateListen
         ) == PackageManager.PERMISSION_GRANTED
     }
 
+    /**
+     * Checks [requestedPermissions] property for permission to determine if it has already been
+     * requested this session. [permissionHasBeenPermanentlyDenied] returns false if we have
+     * never requested such a permission. So it's best to check both of these functions.
+     *
+     * @since 100.6.1
+     */
+    private fun permissionHasBeenRequestedPreviously(permission: String): Boolean =
+        requestedPermissions.contains(permission)
+
     /**
      * Checks if [permission] has been permanently denied.
```

Comment:
I assume this implementation is based on the workaround described here?
https://blog.usejournal.com/method-to-detect-if-user-has-selected-dont-ask-again-while-requesting-for-permission-921b95ded536

Might be worth to to link to it or copy some of it into this doc.

Reply:
I didn't use that solution, no. What would you like me to add? I thought the doc and function name were quite self-explanatory.

Reply:
I suggest adding something like:
"We use ActivityCompat.shouldShowRequestPermissionRationale() as a workaround to detect whether the user has denied the permission permanently, i.e. has selected `Don't Ask Again`. There is no official API to detect that, so we use a variation of a workaround described here: https://blog.usejournal.com/method-to-detect-if-user-has-selected-dont-ask-again-while-requesting-for-permission-921b95ded536"